auth0_flutter/lib/src/web/auth0_flutter_plugin_real.dart

```diff
@@ -1,13 +1,41 @@
-import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart';
+import 'dart:html';
+import 'dart:js_util';
+
+import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart'
+    show Account, Auth0FlutterWebPlatform, LoginOptions;
+
 import 'package:flutter_web_plugins/flutter_web_plugins.dart';
 
-// import 'js_interop.dart';
+import 'js_interop.dart';
 
 class Auth0FlutterPlugin extends Auth0FlutterWebPlatform {
   static void registerWith(final Registrar registrar) {
     Auth0FlutterWebPlatform.instance = Auth0FlutterPlugin();
   }
 
-  // methods, e.g. loginWithRedirect()
-  // here they should call the interop methods, that in turn call the JS methods
+  late Auth0Client client;
+
+  @override
+  Future<void> initialize(final Account account,
+      {final AuthorizationParams? authorizationParams}) async {
+    client = Auth0Client(
+        Auth0ClientOptions(domain: account.domain, clientId: account.clientId));
+
+    final search = window.location.search;
+
+    if (search?.contains('state=') == true &&
+        (search?.contains('code=') == true ||
+            search?.contains('error=') == true)) {
+      await promiseToFuture<void>(client.handleRedirectCallback());
+    } else {
+      await promiseToFuture<void>(client.checkSession());
```

Comment:
Just a note to say that if you want to read the values from resolved promises, you **must** wrap it in `promiseToFuture` to convert it to a Dart Future. Otherwise you just get a pending Promise object back.